pkg/net/libp2p/key_test.go

```diff
@@ -23,25 +24,28 @@ func TestOperatorPrivateKeyToNetworkKeyPair(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if !reflect.DeepEqual(networkPublicKey.Curve.Params(), DefaultCurve.Params()) {
+	ecdsaPublicKey := (*btcec.PublicKey)(networkPublicKey).ToECDSA()
```

Comment:
Please, do not use `ToECDSA` (here and in other test cases below) but `X()` and `Y()` instead. The reasoning is that we shouldn't convert the resulting key to something else, just to perform the assertion on their coordinates.

Reply:
Yep, I think we don't have a choice. You can do it a little bit simpler though:
```
func extractD(privateKey *btcec.PrivateKey) *big.Int {
	privateKeyBytes := privateKey.Key.Bytes()
	return new(big.Int).SetBytes(privateKeyBytes[:])
}
```

Reply:
Done in f9e26c52413369ee9a51d13abaae75564181b1cc.

Note that I had to remove checks for the `Curve` parameter in network public keys, as curve is no longer tied directly to the keys. The `btcec` library uses `secp256k1` curve under the hood, but to access it I would have to call `btcec.S256()` (and not `networkPublicKey.Curve()`), which in my opinion does not make sense.